src/test/app/MultiSign_test.cpp

```diff
@@ -1134,6 +1134,80 @@ class MultiSign_test : public beast::unit_test::suite
         env.close();
     }
 
+    void test_signForHash()
+    {
+        // Make sure that the "hash" field returned by the "sign_for" RPC
+        // command matches the hash returned when that command is sent
+        // through "submit_multisigned".  Make sure that hash also locates
+        // the transaction in the ledger.
+        using namespace jtx;
+        Account const alice {"alice", KeyType::ed25519};
+
+        Env env(*this);
+        env.fund (XRP(1000), alice);
+        env.close();
+
+        env (signers (alice, 2, {{bogie, 1}, {ghost, 1}}));
+        env.close();
+
+        // Use sign_for to sign a transaction where alice pays 10 XRP to
+        // masterpassphrase.
+        std::uint32_t baseFee = env.current()->fees().base;
+        Json::Value jvSig1;
+        jvSig1[jss::account] = bogie.human();
+        jvSig1[jss::secret]  = bogie.name();
+        jvSig1[jss::tx_json][jss::Account]         = alice.human();
+        jvSig1[jss::tx_json][jss::Amount]          = 10000000;
+        jvSig1[jss::tx_json][jss::Destination]     = env.master.human();
+        jvSig1[jss::tx_json][jss::Fee]             = 3 * baseFee;
+        jvSig1[jss::tx_json][jss::Sequence]        = env.seq(alice);
+        jvSig1[jss::tx_json][jss::TransactionType] = "Payment";
+
+        Json::Value jvSig2 = env.rpc (
+            "json", "sign_for", to_string (jvSig1));
+        BEAST_EXPECT (
+            jvSig2[jss::result][jss::status].asString() == "success");
+
+        // Save the hash with one signature for use later.
+        std::string const hash1 =
+            jvSig2[jss::result][jss::tx_json][jss::hash].asString();
+
+        // Add the next signature and sign again.
+        jvSig2[jss::result][jss::account] = ghost.human();
+        jvSig2[jss::result][jss::secret]  = ghost.name();
+        Json::Value jvSubmit = env.rpc (
```

Comment:
Nit: `Json::Value jvSubmit` might be better named `Json::Value jvSig3`.

Reply:
I actually used `jvSig3` initially.  I changed the name because the JSON in `jvSubmit` is what we'll submit to the network (yeah, it's a unit test, so it's a network of one) on line 1190.  So that's my justification for the variable name...